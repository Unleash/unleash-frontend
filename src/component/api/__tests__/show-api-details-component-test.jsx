import React from 'react';

import ShowApiDetailsComponent from '../show-api-details-component';
import renderer from 'react-test-renderer';

test('renders correctly with empty version', () => {
    const uiConfig = {
        name: 'Unleash',
        slogan: 'We are the best!',
        environment: 'test',
        version: '',
    };

    const tree = renderer.create(<ShowApiDetailsComponent uiConfig={uiConfig} />).toJSON();
    expect(tree).toMatchSnapshot();
});

test('renders correctly with ui-config', () => {
    const uiConfig = {
        name: 'Unleash',
        slogan: 'We are the best!',
        environment: 'test',
        version: '1.1.0',
    };

    const tree = renderer.create(<ShowApiDetailsComponent uiConfig={uiConfig} />).toJSON();
    expect(tree).toMatchSnapshot();
});

test('renders correctly without uiConfig', () => {
    const uiConfig = {
        name: 'Unleash',
        version: '1.1.0',
    };

    const tree = renderer.create(<ShowApiDetailsComponent uiConfig={uiConfig} />).toJSON();
    expect(tree).toMatchSnapshot();
});
