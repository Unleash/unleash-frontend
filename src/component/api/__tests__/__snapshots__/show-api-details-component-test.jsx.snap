// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with empty version 1`] = `
<react-mdl-FooterSection
  logo="Unleash "
  type="bottom"
>
  <small>
    (test)
  </small>
  <br />
  <small>
    We are the best!
  </small>
</react-mdl-FooterSection>
`;

exports[`renders correctly with ui-config 1`] = `
<react-mdl-FooterSection
  logo="Unleash 1.1.0"
  type="bottom"
>
  <small>
    (test)
  </small>
  <br />
  <small>
    We are the best!
  </small>
</react-mdl-FooterSection>
`;

exports[`renders correctly without uiConfig 1`] = `
<react-mdl-FooterSection
  logo="Unleash 1.1.0"
  type="bottom"
>
  <small>
    
  </small>
  <br />
  <small />
</react-mdl-FooterSection>
`;
